apps/sports-helsinki/src/domain/search/eventSearch/AdvancedSearch.tsx

```diff
@@ -128,6 +123,60 @@ export const AdvancedSearchForm: React.FC<AdvancedSearchFormProps> = ({
     scrollToResultList();
   };
 
+  return {
+    searchParams,
+    goToSearch,
+    moveToSearchPage,
+    clearInputValues,
+    clearFilters,
+    handleSubmit,
+    initialFieldsOnPageLoad,
+    searchFilters,
+    scrollToResultList,
+  };
+};
+
+/**
+ * @deprecated not in use after the creation of the combined search.
+ * Can be used as an example how to refactor the Hobbies and the Events apps.
+ * */
+export const AdvancedSearchForm: React.FC<AdvancedSearchFormProps> = ({
+  scrollToResultList,
+}) => {
+  const { t } = useTranslation('search');
+  const { t: tAppSports } = useAppSportsTranslation();
+
+  const [autosuggestInput, setAutosuggestInput] = React.useState('');
+
+  const {
+    handleSubmit,
+    clearFilters,
+    goToSearch,
+    initialFieldsOnPageLoad,
+    searchFilters,
+  } = useAdvancedSearchForm({
+    scrollToResultList,
+    autosuggestInput,
+    setAutosuggestInput,
+  });
+
+  const handleMenuOptionClick = async (option: AutosuggestMenuOption) => {
+    const value = option.text;
+
+    const search = getSearchQuery({
+      ...(searchFilters as Filters),
+      text: [value],
+    });
+
+    goToSearch(search);
+    scrollToResultList();
+  };
+
+  // Initialize fields when page is loaded
+  React.useEffect(() => {
+    initialFieldsOnPageLoad();
+  }, [initialFieldsOnPageLoad]);
```

Comment:
Most probably! I think it's just so that my IDE is whining every time about the exhaustive-deps when the dependency array is empty. But, I will change and test it!

Reply:
Just ignore the eslint if this is used once after mounting. One way is to make a separate common hook about it (useMountEffect): 
https://medium.com/@faboulaws/3-custom-react-effect-hooks-for-your-toolbox-96d659d2ca9d